Best practices/Source Generators.md

```diff
@@ -0,0 +1,147 @@
+## What are Source Generators?
+
+According to official [documentation](https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview) a source generator is a part of the .NET Compiler Platform ("Roslyn") SDK which allows developers to generate source files during compilation and add those files to the compilation object. 
```

Comment:
Here we have the info about what they are doing, but we're missing one crucial info: what problems are they solving? When should we use them?